### PR TITLE
tech: Migrate React Router to v6. Remove usage of 'react-router' and use only 'react-router-dom'

### LINKED ISSUE #320: Tech: Move React Router to v6

### PR DESCRIPTION
**Pull request by dmutrokot**
Resolves #320

### CODE REVIEW

File: src/components/RealmSwitchButton/RealmSwitchButton.tsx [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
-    const history = useHistory();
-    const params = qs.parse(location.search);
+    const navigate = useNavigate();
+
+    const { setRealmView } = useContext<any>(ClientContext);
 
     const updateView = () => {
         const path = view === 'list' ? 'map' : 'list';
-        const url = `${match.url}/realm/${path}`;
+        const url = `${path}`;
```
Comment (Vladislav94) [inline src/components/RealmSwitchButton/RealmSwitchButton.tsx:20]: looks like path === url

File: src/pages/Client/ClientRoutes.tsx [hunk 2]
```diff
@@ -87,11 +86,9 @@ export function ClientRoutes() {
                         links={navData}
                         beforeContent={(
                             <Button
-                                to={`/client/${account}`}
+                                to={`${account}`}
```
Comment (Vladislav94) [inline src/pages/Client/ClientRoutes.tsx:89]: seems like here don't need wrap into `${}`

File: src/pages/Client/routes/ClientRealm.tsx [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
         <>
-            <Switch>
-                <Route path={`${match.path}/map`} component={ClientRealmMap} />
-                <Route path={`${match.path}/list`} component={ClientRealmList} />
-                <Redirect from={match.path} to={`${match.path}/${realmView}${location.search}`} />
-            </Switch>
+            <Routes>
+                <Route path='list' element={<ClientRealmList />} />
+                <Route path='map' element={<ClientRealmMap />} />
+                <Route path='*' element={<Navigate to={`${realmView}`} replace />} />
```
Comment (Vladislav94) [inline src/pages/Client/routes/ClientRealm.tsx:17]: seems like here don't need wrap into `${}`

File: src/pages/Raffle/Raffle.tsx [hunk 4]
```diff
@@ -50,8 +49,8 @@ export function Raffle() {
         if (raffleActive && currentRaffle) {
             queryParams.address = raffleActive;
 
-            history.push({
-                pathname: `${match.path}/${currentRaffle.name}`,
+            navigate({
+                pathname: `${currentRaffle.name}`,
```
Comment (Vladislav94) [inline src/pages/Raffle/Raffle.tsx:53]: seems like here don't need wrap into `${}`

File: src/pages/Raffle/Raffle.tsx [hunk 5]
```diff
@@ -82,12 +81,10 @@ export function Raffle() {
...
             <RaffleContextProvider>
-                <Switch>
-                    <Route path={`${match.path}/:name`}>
-                        <RaffleContent user={raffleActive} />
-                    </Route>
-                    <Redirect from='*' to={`${match.path}/${lastRaffle.name}`} />
-                </Switch>
+                <Routes>
+                    <Route path=':name' element={<RaffleContent user={raffleActive} />} />
+                    <Route path='*' element={<Navigate to={`${lastRaffle.name}`} replace /> } />
```
Comment (Vladislav94) [inline src/pages/Raffle/Raffle.tsx:86]: seems like here don't need wrap into `${}`

File: src/pages/Raffle/components/RaffleNav.tsx [hunk 6]
```diff
@@ -24,8 +22,7 @@ export function RaffleNav({ user }: { user: any }) {
                             }
                             component={NavLink}
                             className={classes.button}
-                            activeClassName='active'
-                            to={{ pathname: `${match.url}/${raffle.name}`, search: `?address=${user}` }}
+                            to={{ pathname: `${raffle.name}`, search: `?address=${user}` }}
```
Comment (Vladislav94) [inline src/pages/Raffle/components/RaffleNav.tsx:25]: seems like here don't need wrap into `${}`

File: src/components/PageNav/PageNav.tsx [hunk 7]
```diff
@@ -33,10 +31,7 @@ export function PageNav({ links, beforeContent, afterContent }: PageNavProps) {
                                 startIcon={link.icon}
                                 component={NavLink}
                                 className={classes.button}
-                                activeClassName='active'
-                                to={{
-                                    pathname: `${match.url}/${link.name}`
-                                }}
+                                to={`${link.name}`}
```
Comment (Vladislav94) [inline src/components/PageNav/PageNav.tsx:34]: seems like here don't need wrap into `${}`

Comment (dmutrokot) [reply src/components/PageNav/PageNav.tsx]: Fixed, thanks

File: src/pages/Raffle/components/RaffleNav.tsx (hunk 6 above)
Comment (dmutrokot) [reply src/pages/Raffle/components/RaffleNav.tsx]: Fixed, thanks

File: src/pages/Raffle/Raffle.tsx (hunk 5 above)
Comment (dmutrokot) [reply src/pages/Raffle/Raffle.tsx]: Fixed, thanks

File: src/pages/Raffle/Raffle.tsx (hunk 4 above)
Comment (dmutrokot) [reply src/pages/Raffle/Raffle.tsx]: Fixed, thanks

File: src/pages/Client/routes/ClientRealm.tsx (hunk 3 above)
Comment (dmutrokot) [reply src/pages/Client/routes/ClientRealm.tsx]: Fixed, thanks

File: src/pages/Client/ClientRoutes.tsx (hunk 2 above)
Comment (dmutrokot) [reply src/pages/Client/ClientRoutes.tsx]: Fixed, thanks

File: src/components/RealmSwitchButton/RealmSwitchButton.tsx (hunk 1 above)
Comment (dmutrokot) [reply src/components/RealmSwitchButton/RealmSwitchButton.tsx]: Fixed, thanks

**Review by Vladislav94 — APPROVED** [review]

**Review by dudendy — APPROVED** [review]

**Pull request merged**